### PR TITLE
[node] Update types to v20.6.0

### PR DESCRIPTION
**Pull request by Semigradsky**
https://github.com/nodejs/node/releases/tag/v20.6.0

- `import.meta.resolve` unflagged
- New node:module API `register` for module customization hooks; new `initialize` hook
- inspector: open add SymbolDispose

---
Also:
- fix #66627

### CODE REVIEW

**Comment by Semigradsky** [comment]
Interesting 🤔
```
ERROR: 135:5  expect  TypeScript@4.8 expected type to be:
  "fail" | "ok"
got:
  "ok" | "fail"
```

and

```
ERROR: 135:5  expect  TypeScript@5.3 expected type to be:
  "ok" | "fail"
got:
  "fail" | "ok"
```

File: types/node/async_hooks.d.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ declare module 'async_hooks' {
      * import fs from 'node:fs';
      *
      * console.log(executionAsyncId());  // 1 - bootstrap
+     * const path = '.';
```
Comment (peterblazejewicz) [inline types/node/async_hooks.d.ts:24]: 👍🏻

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@Semigradsky thanks!

**Comment by Semigradsky** [comment]
Ready to merge

**Comment by laverdet** [comment]
@Semigradsky since you broke out the optional parameter of `import.meta.resolve` into a required overload, this fails given `undefined`:
```ts
declare const parent: string | undefined;
import.meta.resolve('specifier', parent);
```
```
Argument of type 'string | undefined' is not assignable to parameter of type 'string | URL'.
  Type 'undefined' is not assignable to type 'string | URL'.ts(2345)
```

Would you be able to submit a fix folding it back in as an optional?

**Comment by Semigradsky** [comment]
@laverdet Sorry for this. Please check https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67431

**Comment by laverdet** [comment]
My intent was not to chastise. He broke a thing: the statement is both neutral and factual. It's also totally acceptable and normal. I break things all the time and I appreciate when people bisect and let me know. I have a ton of respect for type maintainers and I'm proud to be in the same community with Dmitry.

**Comment by laverdet** [comment]
Oh sorry I see we had a miscommunication here. When I used "to break out" I meant like to refactor. Like how when one says "to break out of prison". It can mean a simple refactor of code. I did not mean to say it was an intentional "breakage".

**Pull request merged**